xsd/parse.go

```diff
@@ -176,32 +225,68 @@ func parseType(name xml.Name) Type {
 	return builtin
 }
 
-func anonTypeName(n int, ns string) xml.Name {
-	return xml.Name{ns, fmt.Sprintf("_anon%d", n)}
+func anonTypeName(hash, ns string) xml.Name {
+	return xml.Name{
+		ns,
+		fmt.Sprintf(
+			"_anon_%s",
+			suffixFromHash(hash),
+		),
+	}
 }
 
-/* Convert
-<element name="foo">
-  <complexType>
-  ...
-  </complexType>
-</element>
-
-to
-
-<element name="foo" type="foo">
-  <complexType name="foo">
-  ...
-  </complexType>
-</element>
+/*
+	Convert:
+	<element name="foo">
+	  <complexType>
+	  ...
+	  </complexType>
+	</element>
+
+	to:
+	<element name="foo" type="foo">
+	  <complexType name="foo">
+	  ...
+	  </complexType>
+	</element>
 */
-func copyEltNamesToAnonTypes(root *xmltree.Element) {
-	used := make(map[xml.Name]struct{})
+func prepCopyEltNamesToAnonTypes(
```

Comment:
I'm not a fan of this function name. As time has passed i've tried to structure parsing as a series of discrete passes over the data, where each "pass" is a function that makes a well-defined transformation over the data. And these transformations should be somewhat self-contained.

From the name `prepCopyEltNamesToAnonTypes` I guess this is necessary to run before some CopyEltNamesToAnonTypes stage. That doesn't tell me anything about what this step is for.

I'm still digesting this function so I don't have a good name in mind.

Reply:
I have no issue with big fat functions :) . I vote for moving it into CopyEltNamesToAnonTypes

Reply:
Roger that :-)